dev/devicelab/bin/tasks/fast_scroll_large_images__memory.dart

```diff
@@ -30,7 +30,7 @@ class FastScrollLargeImagesMemoryTest extends MemoryTest {
     await launchApp();
     await recordStart();
     await device.shellExec('input', <String>['swipe', '0 1500 0 0 50']);
-    await Future<void>.delayed(const Duration(milliseconds: 10000));
+    await Future<void>.delayed(const Duration(milliseconds: 15000));
```

Comment:
rather than waiting an arbitrary time, which is super flaky, can we just wait until the test is known to have finished, somehow?

Reply:
(e.g. watching the logs, like some of the other tests do?)

Reply:
Agree that waiting for some actual signal would be much better. Although this would rely on some (to-be-added) signals/logs being generated by the engine in the IO thread, so this would require some engine changes first. There's an additional difficulty that we should wait for the last cleanup signal, but it's unclear to me how to determine whether a signal is the final one as the engine could generate arbitrary number of such signals at arbitrary times. CC @dnfield and @chinmaygarde for more thoughts on how to provide a deterministic signal from the engine.

Reply:
can't we just print from the dart code in the test?

Reply:
When the Dart code released the image, the engine IO thread and Skia would still hold the graphics memory. The actually release won't happen until a `Drain` task is executed, which could be delayed by up to 3 seconds, or even more if there are some slow IO thread tasks. The test only makes sense to measure the memory when the last engine IO thread `Drain` task actually finishes. That's why printing something from the Dart code isn't an option here.